shell/server/core.js

```diff
@@ -140,17 +141,17 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
   }
 
   close() {
```

Comment:
Moreover, I think this is the wrong behavior for `close()`. I believe the logic here should look more like what's in `dropInternal()`. I.e. you should check in `ApiTokens` to see if there's anything left with `frontendRef.notificationHandle` referring to this notification ID, and if not, only then dismiss the notification.